release-25.4: storage: add cluster setting for latency tolerant min size, bump Pebble

annrpom opened the pull request:
Changes:

 * [`65fe99f4`](https://github.com/cockroachdb/pebble/commit/65fe99f4) crl-release-25.4: db: add configurable latency tolerant minimum size
 * [`67287b85`](https://github.com/cockroachdb/pebble/commit/67287b85) db: tolerate unknown options
 * [`b8a41d59`](https://github.com/cockroachdb/pebble/commit/b8a41d59) db: fix RangeDeletionBytesEstimate
 * [`c52566b9`](https://github.com/cockroachdb/pebble/commit/c52566b9) db: add RangeDeletionBytesEstimate blob file test
 * [`b3ea2a92`](https://github.com/cockroachdb/pebble/commit/b3ea2a92) tool/logs: support virtual sst rewrite compaction summary
 * [`3a4bf222`](https://github.com/cockroachdb/pebble/commit/3a4bf222) tool/logs: support blob file rewrites compaction summary

Release note: none.
Epic: none.

---

Release justification: fixes issue where reverting before a major upgrade finalization errors out due to unrecognizable options; adds low-risk tool support for observability; add cluster setting as fail-safe for range local/mvcc garbage separation

---
Add a private cluster setting that controls the minimum size for latency
tolerant (range local keyspace) and likely MVCC keys to be separated into
a blob file.

annrpom committed:
release-25.4: go.mod: bump Pebble to 65fe99f4d031

Changes:

 * [`65fe99f4`](https://github.com/cockroachdb/pebble/commit/65fe99f4) crl-release-25.4: db: add configurable latency tolerant minimum size
 * [`67287b85`](https://github.com/cockroachdb/pebble/commit/67287b85) db: tolerate unknown options
 * [`b8a41d59`](https://github.com/cockroachdb/pebble/commit/b8a41d59) db: fix RangeDeletionBytesEstimate
 * [`c52566b9`](https://github.com/cockroachdb/pebble/commit/c52566b9) db: add RangeDeletionBytesEstimate blob file test
 * [`b3ea2a92`](https://github.com/cockroachdb/pebble/commit/b3ea2a92) tool/logs: support virtual sst rewrite compaction summary
 * [`3a4bf222`](https://github.com/cockroachdb/pebble/commit/3a4bf222) tool/logs: support blob file rewrites compaction summary

Release note: none.
Epic: none.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/156556)
<!-- Reviewable:end -->

annrpom committed:
release-25.4: storage: add cluster setting for latency tolerant min size

Add a private cluster setting that controls the minimum size for latency
tolerant (range local keyspace) and likely MVCC keys to be separated into
a blob file.

Release note: None
Epic: none

jbowens reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> @jbowens reviewed 4 of 4 files at r1, 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/156556#-:-Ocv58Qu3QUFt92Cbmz3:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @xinhaoz)



<!-- Sent from Reviewable.io -->
